File: pkg/flutter/doctor.go
package flutter

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"

	"github.com/clix-so/clix-cli/pkg/logx"
	"github.com/clix-so/clix-cli/pkg/utils"
	"github.com/clix-so/clix-cli/pkg/versions"
)

// RunDoctor checks the Flutter project setup for Clix SDK
func RunDoctor() error {
	projectRoot, err := os.Getwd()
	if err != nil {
		return fmt.Errorf("failed to get current working directory: %v", err)
	}

	fmt.Println("🏥 Clix Flutter Doctor")
	logx.Separatorln()

	allChecks := []struct {
		name string
		fn   func(string) (bool, string)
	}{
		{"Flutter Project Detection", checkFlutterProject},
		{"Firebase CLI Installation", checkFirebaseCLI},
		{"FlutterFire CLI Installation", checkFlutterFireCLI},
		{"Firebase Options Configuration", checkFirebaseOptions},
		{"Clix Flutter SDK Dependency", checkClixDependency},
		{"Firebase Core Dependency", checkFirebaseCoreDependency},
		{"Firebase Messaging Dependency", checkFirebaseMessagingDependency},
		{"Main.dart Configuration", checkMainDartConfiguration},
	}

	allPassed := true

	for _, check := range allChecks {
		passed, message := check.fn(projectRoot)
		if passed {
			logx.Log().Branch().Success().Println(fmt.Sprintf("%s ✅", check.name))
		} else {
			logx.Log().Branch().Failure().Println(fmt.Sprintf("%s ❌", check.name))
			if message != "" {
				logx.Log().Indent(6).Code().Println(message)
			}
			allPassed = false
		}
	}

	logx.NewLine()

	if allPassed {
		fmt.Println("🎉 All checks passed! Your Flutter project is properly configured for Clix SDK.")
	} else {
		fmt.Println("❗ Some checks failed. Please fix the issues above and run 'clix doctor --flutter' again.")
		return fmt.Errorf("flutter doctor checks failed")
	}

	return nil
}

// checkFlutterProject verifies this is a Flutter project
func checkFlutterProject(projectRoot string) (bool, string) {
	pubspecPath := filepath.Join(projectRoot, "pubspec.yaml")
	if _, err := os.Stat(pubspecPath); err != nil {
		return false, "pubspec.yaml not found - not a Flutter project"
	}

	data, err := os.ReadFile(pubspecPath)
	if err != nil {
		return false, "failed to read pubspec.yaml"
	}

	content := string(data)
	if !strings.Contains(content, "flutter:") {
		return false, "flutter dependency not found in pubspec.yaml"
	}

	return true, ""
}

// checkClixDependency verifies Clix Flutter SDK dependency
func checkClixDependency(projectRoot string) (bool, string) {
	pubspecPath := filepath.Join(projectRoot, "pubspec.yaml")
	data, err := os.ReadFile(pubspecPath)
	if err != nil {
		return false, "failed to read pubspec.yaml"
	}

	content := string(data)
	if !strings.Contains(content, "clix_flutter:") {
		return false, "Add 'clix_flutter: " + versions.FlutterClixSDKVersion + "' to dependencies in pubspec.yaml"
	}

	return true, ""
}

// checkFirebaseCoreDependency verifies Firebase Core dependency
func checkFirebaseCoreDependency(projectRoot string) (bool, string) {
	pubspecPath := filepath.Join(projectRoot, "pubspec.yaml")
	data, err := os.ReadFile(pubspecPath)
	if err != nil {
		return false, "failed to read pubspec.yaml"
	}

	content := string(data)
	if !strings.Contains(content, "firebase_core:") {
		return false, "Add 'firebase_core: " + versions.FlutterFirebaseCoreVersion + "' to dependencies in pubspec.yaml"
	}

	return true, ""
}

// checkFirebaseMessagingDependency verifies Firebase Messaging dependency
func checkFirebaseMessagingDependency(projectRoot string) (bool, string) {
	pubspecPath := filepath.Join(projectRoot, "pubspec.yaml")
	data, err := os.ReadFile(pubspecPath)
	if err != nil {
		return false, "failed to read pubspec.yaml"
	}

	content := string(data)
	if !strings.Contains(content, "firebase_messaging:") {
		return false, "Add 'firebase_messaging: " + versions.FlutterFirebaseMessagingVersion + "' to dependencies in pubspec.yaml"
	}

	return true, ""
}

// checkFirebaseCLI verifies Firebase CLI is installed
func checkFirebaseCLI(projectRoot string) (bool, string) {
	if err := utils.RunShellCommand("firebase", "--version"); err != nil {
		return false, "Firebase CLI not installed. Run: npm install -g firebase-tools"
	}
	return true, ""
}

// checkFlutterFireCLI verifies FlutterFire CLI is installed
func checkFlutterFireCLI(projectRoot string) (bool, string) {
	if err := utils.RunShellCommand("flutterfire", "--version"); err != nil {
		return false, "FlutterFire CLI not installed. Run: dart pub global activate flutterfire_cli"
	}
	return true, ""
}

// checkFirebaseOptions verifies firebase_options.dart exists
func checkFirebaseOptions(projectRoot string) (bool, string) {
	configPath := filepath.Join(projectRoot, "lib", "firebase_options.dart")
	if _, err := os.Stat(configPath); err != nil {
		return false, "firebase_options.dart not found. Run: flutterfire configure"
	}

	// Check if Firebase config files exist (they should be created by flutterfire configure)
	androidConfigPath := filepath.Join(projectRoot, "android", "app", "google-services.json")
	iosConfigPath := filepath.Join(projectRoot, "ios", "Runner", "GoogleService-Info.plist")

	var missingFiles []string
	if _, err := os.Stat(androidConfigPath); err != nil {
		missingFiles = append(missingFiles, "android/app/google-services.json")
	}
	if _, err := os.Stat(iosConfigPath); err != nil {
		missingFiles = append(missingFiles, "ios/Runner/GoogleService-Info.plist")
	}

	if len(missingFiles) > 0 {
		return false, fmt.Sprintf("missing Firebase config files: %v. Run: flutterfire configure", missingFiles)
	}

	return true, ""
}

// checkMainDartConfiguration verifies main.dart has proper Clix setup
func checkMainDartConfiguration(projectRoot string) (bool, string) {
	mainPath := filepath.Join(projectRoot, "lib", "main.dart")
	if _, err := os.Stat(mainPath); err != nil {
		return false, "main.dart not found at lib/main.dart"
	}

	data, err := os.ReadFile(mainPath)
	if err != nil {
		return false, "failed to read main.dart"
	}

	content := string(data)
	issues := []string{}

	if !strings.Contains(content, "firebase_core") {
		issues = append(issues, "Missing firebase_core import")
	}

	if !strings.Contains(content, "firebase_options.dart") {
		issues = append(issues, "Missing firebase_options.dart import")
	}

	if !strings.Contains(content, "clix_flutter") {
		issues = append(issues, "Missing clix_flutter import")
	}

	if !strings.Contains(content, "Firebase.initializeApp") {
		issues = append(issues, "Missing Firebase.initializeApp call")
	}

	if !strings.Contains(content, "DefaultFirebaseOptions.currentPlatform") {
		issues = append(issues, "Missing DefaultFirebaseOptions.currentPlatform in Firebase.initializeApp")
	}

	if !strings.Contains(content, "Clix.initialize") {
		issues = append(issues, "Missing Clix.initialize call")
	}

	if !strings.Contains(content, "WidgetsFlutterBinding.ensureInitialized()") {
		issues = append(issues, "Missing WidgetsFlutterBinding.ensureInitialized() call")
	}

	if len(issues) > 0 {
		return false, strings.Join(issues, "; ")
	}

	return true, ""
}

File: src/packages/flutter/doctor.ts
import { promises as fs } from 'fs';
import path from 'path';
import { runCommand } from '../../utils/shell.js';
import {
  FLUTTER_CLIX_SDK_VERSION,
  FLUTTER_FIREBASE_CORE_VERSION,
  FLUTTER_FIREBASE_MESSAGING_VERSION,
} from '../versions.js';

export async function runFlutterDoctor(): Promise<void> {
  const projectRoot = process.cwd();

  console.log('🏥 Clix Flutter Doctor');
  console.log('==================================================\n');

  const allChecks: Array<{
    name: string;
    fn: (projectRoot: string) => Promise<{ passed: boolean; message: string }>;
  }> = [
    { name: 'Flutter Project Detection', fn: checkFlutterProject },
    { name: 'Firebase CLI Installation', fn: checkFirebaseCLI },
    { name: 'FlutterFire CLI Installation', fn: checkFlutterFireCLI },
    { name: 'Firebase Options Configuration', fn: checkFirebaseOptions },
    { name: 'Clix Flutter SDK Dependency', fn: checkClixDependency },
    { name: 'Firebase Core Dependency', fn: checkFirebaseCoreDependency },
    { name: 'Firebase Messaging Dependency', fn: checkFirebaseMessagingDependency },
    { name: 'Main.dart Configuration', fn: checkMainDartConfiguration },
  ];

  let allPassed = true;

  for (const check of allChecks) {
    const { passed, message } = await check.fn(projectRoot);
    if (passed) {
      console.log(`✅ ${check.name}`);
    } else {
      console.log(`❌ ${check.name}`);
      if (message) {
        console.log(`   ${message}`);
      }
      allPassed = false;
    }
  }

  console.log('');

  if (allPassed) {
    console.log('🎉 All checks passed! Your Flutter project is properly configured for Clix SDK.');
  } else {
    console.log("❗ Some checks failed. Please fix the issues above and run 'clix doctor --flutter' again.");
  }
}

async function checkFlutterProject(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const pubspecPath = path.join(projectRoot, 'pubspec.yaml');
  try {
    await fs.stat(pubspecPath);
  } catch {
    return { passed: false, message: 'pubspec.yaml not found - not a Flutter project' };
  }

  try {
    const data = await fs.readFile(pubspecPath, 'utf-8');
    if (!data.includes('flutter:')) {
      return { passed: false, message: 'flutter dependency not found in pubspec.yaml' };
    }
  } catch {
    return { passed: false, message: 'failed to read pubspec.yaml' };
  }

  return { passed: true, message: '' };
}

async function checkClixDependency(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const pubspecPath = path.join(projectRoot, 'pubspec.yaml');
  try {
    const data = await fs.readFile(pubspecPath, 'utf-8');
    if (!data.includes('clix_flutter:')) {
      return {
        passed: false,
        message: `Add 'clix_flutter: ${FLUTTER_CLIX_SDK_VERSION}' to dependencies in pubspec.yaml`,
      };
    }
  } catch {
    return { passed: false, message: 'failed to read pubspec.yaml' };
  }

  return { passed: true, message: '' };
}

async function checkFirebaseCoreDependency(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const pubspecPath = path.join(projectRoot, 'pubspec.yaml');
  try {
    const data = await fs.readFile(pubspecPath, 'utf-8');
    if (!data.includes('firebase_core:')) {
      return {
        passed: false,
        message: `Add 'firebase_core: ${FLUTTER_FIREBASE_CORE_VERSION}' to dependencies in pubspec.yaml`,
      };
    }
  } catch {
    return { passed: false, message: 'failed to read pubspec.yaml' };
  }

  return { passed: true, message: '' };
}

async function checkFirebaseMessagingDependency(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const pubspecPath = path.join(projectRoot, 'pubspec.yaml');
  try {
    const data = await fs.readFile(pubspecPath, 'utf-8');
    if (!data.includes('firebase_messaging:')) {
      return {
        passed: false,
        message: `Add 'firebase_messaging: ${FLUTTER_FIREBASE_MESSAGING_VERSION}' to dependencies in pubspec.yaml`,
      };
    }
  } catch {
    return { passed: false, message: 'failed to read pubspec.yaml' };
  }

  return { passed: true, message: '' };
}

async function checkFirebaseCLI(_projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const result = await runCommand('firebase', ['--version']);
  if (!result.success) {
    return { passed: false, message: 'Firebase CLI not installed. Run: npm install -g firebase-tools' };
  }
  return { passed: true, message: '' };
}

async function checkFlutterFireCLI(_projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const result = await runCommand('flutterfire', ['--version']);
  if (!result.success) {
    return { passed: false, message: 'FlutterFire CLI not installed. Run: dart pub global activate flutterfire_cli' };
  }
  return { passed: true, message: '' };
}

async function checkFirebaseOptions(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const configPath = path.join(projectRoot, 'lib', 'firebase_options.dart');
  try {
    await fs.stat(configPath);
  } catch {
    return { passed: false, message: 'firebase_options.dart not found. Run: flutterfire configure' };
  }

  // Check if Firebase config files exist (they should be created by flutterfire configure)
  const androidConfigPath = path.join(projectRoot, 'android', 'app', 'google-services.json');
  const iosConfigPath = path.join(projectRoot, 'ios', 'Runner', 'GoogleService-Info.plist');

  const missingFiles: string[] = [];
  try {
    await fs.stat(androidConfigPath);
  } catch {
    missingFiles.push('android/app/google-services.json');
  }
  try {
    await fs.stat(iosConfigPath);
  } catch {
    missingFiles.push('ios/Runner/GoogleService-Info.plist');
  }

  if (missingFiles.length > 0) {
    return { passed: false, message: `missing Firebase config files: ${missingFiles.join(', ')}. Run: flutterfire configure` };
  }

  return { passed: true, message: '' };
}

async function checkMainDartConfiguration(projectRoot: string): Promise<{ passed: boolean; message: string }> {
  const mainPath = path.join(projectRoot, 'lib', 'main.dart');
  try {
    await fs.stat(mainPath);
  } catch {
    return { passed: false, message: 'main.dart not found at lib/main.dart' };
  }

  try {
    const data = await fs.readFile(mainPath, 'utf-8');
    const content = data;
    const issues: string[] = [];

    if (!content.includes('firebase_core')) {
      issues.push('Missing firebase_core import');
    }

    if (!content.includes('firebase_options.dart')) {
      issues.push('Missing firebase_options.dart import');
    }

    if (!content.includes('clix_flutter')) {
      issues.push('Missing clix_flutter import');
    }

    if (!content.includes('Firebase.initializeApp')) {
      issues.push('Missing Firebase.initializeApp call');
    }

    if (!content.includes('DefaultFirebaseOptions.currentPlatform')) {
      issues.push('Missing DefaultFirebaseOptions.currentPlatform in Firebase.initializeApp');
    }

    if (!content.includes('Clix.initialize')) {
      issues.push('Missing Clix.initialize call');
    }

    if (!content.includes('WidgetsFlutterBinding.ensureInitialized()')) {
      issues.push('Missing WidgetsFlutterBinding.ensureInitialized() call');
    }

    if (issues.length > 0) {
      return { passed: false, message: issues.join('; ') };
    }

    return { passed: true, message: '' };
  } catch {
    return { passed: false, message: 'failed to read main.dart' };
  }
}
